src/backend/joanie/core/serializers.py

```diff
@@ -1,28 +1,183 @@
+        - targeted courses with its course runs
+            - If user is authenticated, we try to retrieve enrollment related
+              to each course runs.
+        - order if user is authenticated
     """
 
     id = serializers.CharField(source="uid", read_only=True)
+    certificate = CertificationDefinitionSerializer(
+        read_only=True, source="certificate_definition"
+    )
+    currency = serializers.SerializerMethodField()
+    order = serializers.SerializerMethodField(read_only=True)
+    price = serializers.DecimalField(
+        max_digits=9,
+        decimal_places=2,
+        coerce_to_string=False,
+        min_value=0,
+    )
+    target_courses = serializers.SerializerMethodField()
 
     class Meta:
         model = models.Product
-        fields = ["id", "title", "call_to_action", "price"]
+        fields = [
+            "call_to_action",
+            "certificate",
+            "currency",
+            "id",
+            "order",
+            "price",
+            "target_courses",
+            "title",
+            "type",
+        ]
+
+    def get_target_courses(self, product):
+        """
+        For the current product, retrieve its related courses.
+        """
+
+        context = self.context.copy()
+        context.update({"product": product})
+
+        return TargetCourseSerializer(
+            instance=models.Course.objects.filter(
+                product_relations__product=product
+            ).order_by("product_relations__position"),
+            many=True,
+            context=context,
+        ).data
+
+    def get_currency(self, _):  # pylint: disable=no-self-use
+        """
+        Return currency code and symbol
+        to allow frontend to format properly price
+        """
+        (code, symbol) = settings.JOANIE_CURRENCY[:2]
+        return {"code": code, "symbol": symbol}
+
+    def get_order(self, product):
+        """
+        For the current product, retrieve valid order related
+        to the authenticated user.
+        """
+        user = self.context.get("user", None)
+
+        if isinstance(user, get_user_model()):
+            try:
+                order = models.Order.objects.get(
+                    product=product,
+                    owner=user,
+                    course__code=self.context["code"],
+                    state__in=[enums.ORDER_STATE_PAID, enums.ORDER_STATE_FINISHED],
+                )
+
+                return OrderLiteSerializer(order).data
+
+            except ObjectDoesNotExist:
+                pass
+
+        return None
 
 
 class CourseSerializer(serializers.ModelSerializer):
     """
     Serialize all information about a course.
     """
 
-    organization = serializers.CharField(source="organization.code", read_only=True)
-    products = serializers.SlugRelatedField(many=True, read_only=True, slug_field="uid")
+    organization = OrganizationSerializer(read_only=True)
+    products = ProductSerializer(read_only=True, many=True)
```

Comment:
Now that we are building such a complex object, we need to activate cache on the API because this will be called each time a syllabus is displayed?

Reply:
ouch that's a lot of keys to cache...what about caching the enrolment information in the frontend? It only changes via actions from the frontend right?

Reply:
Ok, let's cache only response when user is not authenticated. From Richie, frontend is able to cache information for authenticated user.